### DISCUSSION TITLE
OpenNEM API Registration

### DISCUSSION THREAD

**Discussion by nc9** [opennem/opennem#243] [General]
Starting in April and with our next release the OpenNEM API will be placed behind authorization and will require
an API key to access. We will be providing a free level of access for researchers.

We plan on gradually and completely deprecating the current API that is open to all.

If you are a current API user, or a potential API user, [please fill in this form](https://docs.google.com/forms/d/e/1FAIpQLSf3yOzyrmYJJLN6NYlc15Kt3IPCNQiF-Gv5e9PwvAyg5tRsAQ/viewform?usp=sf_link) to register for access to the API. We will be in touch with you with an email with the various setup steps and access details.

We would like to hear from users what they may use the API for, and what they would like to see from the API.

**Comment by engwithrosie** [comment]
I am hoping to use the API to get access to historical data to make charts and simple models mostly for communication purposes (I have a YouTube channel on clean energy).
I have made a couple of requests for access to the API over the past few weeks and haven't heard anything back. Am I doing something wrong, or perhaps it's due to time off over Christmas?

**Reply by vivek-kasar38** [reply]
Hi Team, 
just looping in here - could you also please get back to me aswell. Thank you

**Reply by wrube** [reply]
> I am hoping to use the API to get access to historical data to make charts and simple models mostly for communication purposes (I have a YouTube channel on clean energy). I have made a couple of requests for access to the API over the past few weeks and haven't heard anything back. Am I doing something wrong, or perhaps it's due to time off over Christmas?

Hi, I have also filled in the linked form to  request the API access key several weeks ago. Is this something OpenNEM  is still offering?

Thanks

Ben

**Reply by nc9** [reply]
Hi Ben - see my comment below. If you go an signup for the waitlist at https://platform.openelectricity.org.au I'll send you an invitation to try it out.

**Comment by AndrewStyan** [comment]
Hi folks,
I requested API access a couple of weeks back but have heard nothing. Is this still available?

I am a visual artist developing an immersive interactive 3D media art platform for exploring system data visualisations - systems being any physical, biological or social system at planetary or local scale. This is the subject of my recent PhD and intended to foster 'system thinking'. The project, called You Are Here, is [described here](https://andrewstyan.com/you-are-here/).

Electricity grid and generation is a clear 'social system' with connections with many other social, physical and biological systems and I have been developing a visualisation. Progress so far is shown below, bars colour coded by fuel, with data from the earlier OpenNEM API. The grid is sourced from open street map, colour coded by voltage. The style is still being fine tuned but it will be animated based on 24 hr history data. The many red bars are where history is not available via the OpenNEM API, but is definitely available as it appears on the OpenElectricity tracker.

Can you progress my application please? 

Thanks,
Andrew Styan

<img width="1267" alt="Screenshot 2025-02-15 at 5 40 33 pm" src="https://github.com/user-attachments/assets/1e937629-eb9f-486e-84f6-246d9e7fd02d" />

<img width="1547" alt="Screenshot 2025-02-15 at 5 42 32 pm" src="https://github.com/user-attachments/assets/68019d71-2112-4cc2-8084-fdfd275840de" />

**Comment by nc9** [comment]
Hi - anybody here requesting API access, go to:

https://platform.openelectricity.org.au/

and signup for the waitlist - I'll send you out an invitation once you do. We're letting people in steadily and it's all a bit in beta at the moment, but you should be able to access the API and data and use our two client libraries (typescript + python)

**Reply by AndrewStyan** [reply]
Thanks Nik, just registered.

**Comment by wrube** [comment]
Thanks so much! I've joined the waiting list too with my university email.

Cheers

Ben

**Comment by dominicboston-kaluza** [comment]
Just for clarification. This link still works and has data, but the data is stale. So authentication doesn't seem to be needed for this... but the data is old. 

https://data.openelectricity.org.au/v3/stats/au/NEM/VIC1/power/7d.json
